### PR TITLE
C++ implmentation of addition using bits

### PR DESCRIPTION
**Pull request by Dhenderson10**
**Fixes issue:**

Issue [#3408](https://github.com/OpenGenus/cosmos/issues/3408)

**Changes:**

Added addition_using_bits.cpp

### CODE REVIEW

File: code/bit_manipulation/src/addition_using_bits/addition_using_bits.cpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#include <iostream>
+#include <string>
+
+using namespace std;
```
Comment (arnavb) [inline code/bit_manipulation/src/addition_using_bits/addition_using_bits.cpp]: Don't do this.

File: code/bit_manipulation/src/addition_using_bits/addition_using_bits.cpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        int i = n & m;
+        return bitwiseAdditionRecursive(n ^ m, i << 1);
+    }
+}
+
+int main()
+{
+    int n = 10;
+    int m = 5;
+    cout << bitwiseAddition(n, m) << endl;
```
Comment (arnavb) [inline code/bit_manipulation/src/addition_using_bits/addition_using_bits.cpp]: Use `\n` instead.

File: code/bit_manipulation/src/addition_using_bits/addition_using_bits.cpp [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    }
+}
+
+int main()
+{
+    int n = 10;
+    int m = 5;
+    cout << bitwiseAddition(n, m) << endl; 
+    cout << bitwiseAdditionRecursive(n, m) << endl;
+    return 0;
```
Comment (arnavb) [inline code/bit_manipulation/src/addition_using_bits/addition_using_bits.cpp]: Implicit; Not needed.

File: code/bit_manipulation/src/addition_using_bits/addition_using_bits.cpp [hunk 4]
```diff
@@ -0,0 +1,37 @@
+#include <iostream>
+#include <string>
```
Comment (arnavb) [inline code/bit_manipulation/src/addition_using_bits/addition_using_bits.cpp]: Any reason this is included?

Comment (Dhenderson10) [reply code/bit_manipulation/src/addition_using_bits/addition_using_bits.cpp]: Nope, left it there from working on something else. Added fixes, let me know if anything else should be changed.

**Comment by Dhenderson10** [comment]
@arnavb Awesome! Thank you!

**Comment by arnavb** [comment]
Hey, on a side note, I would reconfigure your git email address, as it doesn't match the one for your profile. Otherwise, you won't get credit for your commits! 😝

**Comment by Dhenderson10** [comment]
@arnavb I want that credit. Thanks for the heads up! :)

**Comment by arnavb** [comment]
@iattempt Please review!

File: code/bit_manipulation/src/addition_using_bits/addition_using_bits.cpp [hunk 5]
```diff
@@ -0,0 +1,33 @@
+#include <iostream>
+
+// Part of Cosmos by OpenGenus Foundation
+int bitwiseAddition(int n, int m)
+{
+    while (m != 0)
+    {
+        int i = n & m;  // Variable i keeps track of bits that carry over
```
Comment (abdouskamel) [inline code/bit_manipulation/src/addition_using_bits/addition_using_bits.cpp]: This variable should be declared outside the loop.

Comment (arnavb) [reply code/bit_manipulation/src/addition_using_bits/addition_using_bits.cpp]: @abdouskamel Why?

Comment (iattempt) [reply code/bit_manipulation/src/addition_using_bits/addition_using_bits.cpp]: @Dhenderson10 Could you rename i to carry?

File: code/bit_manipulation/src/addition_using_bits/addition_using_bits.cpp [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    }
+}
+
+int main()
+{
+    int n = 10;
+    int m = 5;
+    std::cout << bitwiseAddition(n, m) << "\n"; 
+    std::cout << bitwiseAdditionRecursive(n, m) << "\n";
+}
```
Comment (iattempt) [inline code/bit_manipulation/src/addition_using_bits/addition_using_bits.cpp]: Please use assertion with built-in plus operator and separate src/test files.

Comment (Dhenderson10) [reply code/bit_manipulation/src/addition_using_bits/addition_using_bits.cpp]: @iattempt Taken care of, let me know if anything else needs to be changed.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**